### PR TITLE
feat: add decolorize processor

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Implement the `decolorize` processor similar to the one from Loki https://grafana.com/docs/loki/latest/query/log_queries/#removing-color-codes

Also remove some dead code BTW

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

File: src/pipeline/src/etl/processor/cmcd.rs [hunk 1]
```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Pipeline Processer for CMCD (Common Media Client Data) data.
```
Comment (waynexia) [inline src/pipeline/src/etl/processor/cmcd.rs:15]:
```suggestion
//! Pipeline Processor for CMCD (Common Media Client Data) data.
```

**Comment by paomian** [comment]
Can we add some tests that generate pipelines from yaml and verify that they are correct？like test in `src/pipeline/tests/pipeline.rs` And it can be used as an example to show others how to use it.

**Comment by waynexia** [comment]
> Can we add some tests that generate pipelines from yaml and verify that they are correct？like test in `src/pipeline/tests/pipeline.rs` And it can be used as an example to show others how to use it.

add one in bcd084f

**Review by shuiyisong — APPROVED** [review]
LGTM

**Review by paomian — APPROVED** [review]

**Pull request merged**